### PR TITLE
[Merged by Bors] - feat (RingTheory/HahnSeries) SMulZeroClass instances

### PR DESCRIPTION
**Pull request by ScottCarnahan**
This PR adds instances for `SMulZeroClass R (HahnSeries Γ R)` and `SMulZeroClass R (HahnModule Γ R V)`.  Also, some minor variable cleanup.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/HahnSeries/Multiplication.lean [hunk 1]
```diff
@@ -80,25 +80,26 @@ def HahnModule (Γ R V : Type*) [PartialOrder Γ] [Zero V] [SMul R V] :=
 
 namespace HahnModule
 
-section
-
-variable {Γ R V : Type*} [PartialOrder Γ] [Zero V] [SMul R V]
```
Comment (jcommelin) [inline Mathlib/RingTheory/HahnSeries/Multiplication.lean:85]: What is the reason for inlining these typeclass assumptions?

**Review by jcommelin — COMMENTED** [review]
Looks good to me. Minor question. Feel free to address it however you seem fit.

bors d+

File: Mathlib/RingTheory/HahnSeries/Multiplication.lean (hunk 1 above)
Comment (ScottCarnahan) [reply Mathlib/RingTheory/HahnSeries/Multiplication.lean:85]: I cannot remember any particular reason.  I have restored the variable declaration inside a small section.

**Pull request closed without merging**